src/librustc/ty/context.rs

```diff
@@ -923,7 +923,7 @@ pub struct GlobalCtxt<'tcx> {
 
     maybe_unused_trait_imports: FxHashSet<DefId>,
     maybe_unused_extern_crates: Vec<(DefId, Span)>,
-    pub extern_prelude: FxHashSet<ast::Name>,
+    pub extern_prelude: FxHashMap<ast::Name, bool /* introduced by item */>,
```

Comment:
I'd prefer to see this as a `///`-style comment -- and maybe a bit more spelled out what it means =)